cmds/networkd/addr_watcher.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"context"
+
+	"github.com/rs/zerolog/log"
+
+	"time"
+
+	"github.com/cenkalti/backoff/v3"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/network"
+	"github.com/threefoldtech/zos/pkg/network/ifaceutil"
+	"github.com/threefoldtech/zos/pkg/network/types"
+	"github.com/vishvananda/netlink"
+)
+
+type WatchedLinks struct {
+	linkNames map[string]struct{}
+	bcdb      network.TNoDB
+	nodeID    pkg.Identifier
+}
+
+func NewWatchedLinks(linkNames []string, nodeID pkg.Identifier, db network.TNoDB) WatchedLinks {
+	names := make(map[string]struct{}, len(linkNames))
+
+	for _, n := range linkNames {
+		names[n] = struct{}{}
+	}
+
+	return WatchedLinks{
+		linkNames: names,
+		bcdb:      db,
+		nodeID:    nodeID,
+	}
+}
+
+func (w WatchedLinks) CallBack(update netlink.AddrUpdate) error {
+	link, err := netlink.LinkByIndex(update.LinkIndex)
+	if err != nil {
+		return nil
+	}
+
+	// skip link that are not watched
+	if _, ok := w.linkNames[link.Attrs().Name]; !ok {
+		return nil
+	}
+
+	log.Info().Msg("send network interfaces update to BCDB")
+
+	ifaces, err := getLocalInterfaces()
+	if err != nil {
+		return err
+	}
+
+	return publishIfaces(ifaces, w.nodeID, w.bcdb)
+}
+
+func getLocalInterfaces() ([]types.IfaceInfo, error) {
+	var output []types.IfaceInfo
+
+	links, err := netlink.LinkList()
+	if err != nil {
+		log.Error().Err(err).Msgf("failed to list interfaces")
+		return nil, err
+	}
+
+	for _, link := range ifaceutil.LinkFilter(links, []string{"device", "bridge"}) {
+		//TODO: why bringing the interface up? shouldn't we just check if it's up or not?
```

Comment:
a NIC of which the MII has no handshake detected, doesn't matter if it's up or down, so we bring them up, in case there is some IPv6 RA on that link.